tutorial/advanced/t15_finetuning.ipynb

```diff
@@ -663,7 +654,11 @@
     "        x = fn.relu(self.backbone_layers[2](x))\n",
     "        return x\n",
     "\n",
-    "backbone_torch = fe.build(model_fn=lambda: BackboneTorch(model_torch_pretrained), optimizer_fn=\"adam\") "
+    "\n",
+    "if isinstance(model_torch_pretrained, torch.nn.DataParallel):\n",
+    "    backbone_torch = fe.build(model_fn=lambda: BackboneTorch(model_torch_pretrained.module), optimizer_fn=\"adam\") \n",
```

Comment:
is there a way we can do the backend function where the end user doesn't have to make this if-check?

Reply:
We can do it if we don't use the fe.backend.load_model function in the previous step.

Reply:
why did it used to work without this consideration? or was it an existing bug?

Reply:
@TortoiseHam 
1) Previous implementation directly loads the weight file:
model_torch_pretrained = lenet_torch(input_shape=torch_input_shape, classes=base_num_classes)
model_torch_pretrained.load_state_dict(load(torch_pretrained_model_path))
This would fail, if the weights file previously saved from multi gpu machine directly without using fe.backend.save_model(current implementation) which would introduce "module" prefix into the keys.

2) To avoid this we are trying to use fe.backend.load_model which handles the module inconsistency, but prerequisite is it should be a fe.build.

3) But using fe.build model on multi gpu machine would generate a dataparallel module, which would create a new problem while modifying the model architecture for fine tuning, because of which we need to add if condition to handle that.

Reply:
maybe this if check would be better to put inside the BackboneTorch class init since that would show the better practice for anyone who was designing based on the tutorial